QA. Merge into `trunk`

mishautkin opened the pull request:
### Description

`qa` branch contains numerous updates for
- automated env setup scripts;
- `/tests/qa` dir with autotests;
- `/tests/qa-legacy-ui` dir with autotests.

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa-pay-later-messaging

mishautkin committed:
Merge pull request #3184 from woocommerce/qa-pay-later-messaging

QA Update PLM

mishautkin committed:
Add PLM default UI test

mishautkin committed:
Merge pull request #3187 from woocommerce/qa-plm-default-ui

Add PLM default UI test

mishautkin committed:
Add test `Settings - Pay Later Messaging - Disabled on all pages`

mishautkin committed:
Merge pull request #3188 from woocommerce/qa-pml-disable-on-all-pages

QA - Add test `Settings - Pay Later Messaging - Disabled on all pages`

mishautkin committed:
Add tests for PLM configurator

- Add test data file
- Add separate subtests for each data variation
- Add resulting tests according to Xray
- Add helper methods to save and retreive results of subtests

mishautkin committed:
Fix beforeAll, function name and lint issues

mishautkin committed:
Merge pull request #3191 from woocommerce/qa-plm-configurator-tests

Add tests for PLM configurator

Commit:
Fix learn more links - Italy

Commit:
Settings - Onboarding - Default UI test updated.

* A function has been created to check and verify the console messages for badgeBoxUtils.js
* Test data has been created badges-per-country.data.ts file
* Test case updated to test countries included in the test data
* Fix lint issues

Commit:
fix: var name changed

Commit:
.only removed

Commit:
fix: the test data file name changed

Commit:
fix: var name changed in spec file

Commit:
Merge pull request #3206 from woocommerce/qa-update-default-ui-test

Settings - Onboarding - Default UI test updated.

Commit:
Fix and updates

* Test for Croatia and UK removed as it is included in the first test block.
* Test data for Croatia and UK removed from test data file.
* Test keys updated with actual test keys from Jira

Commit:
camelCase used in var name

Commit:
Unsued import removed

Commit:
Update in functions

* To have a more generic approach for enabling and disabling the advanced connection options enableConnectionOptionsToggle() function created.
* toggleManuallyConnect and toggleSandboxMode update accordingly.

Commit:
Merge pull request #3209 from woocommerce/sedat-work

QA- updates and fixes

Commit:
A missing step added to Learn more links test case

Commit:
Updates

* Test cases in serial block reordered.
* Some steps added to the test cases to prevent failure.

Commit:
Merge pull request #3215 from woocommerce/qa-update-visual-default-ui-test

Qa - update visual default UI test

Commit:
Add test PCP-4315 and PCP-4316.

Some tests were removed as they are going to be automated under PCP-4318

Commit:
Add test PCP-4315 and PCP-4316.

Some tests were removed as they are going to be automated under PCP-4318

Commit:
Fix lint issues

Commit:
Fix lint issues

Commit:
Add test and remove unnecessary ones

Add test PCP-4318
Remove the tests that are in scope of PCP-4318

Commit:
Fix lint issues

Commit:
Merge pull request #3220 from woocommerce/qa-pcp-4315/4316

QA/Add test pcp 4315/4316

Commit:
Merge pull request #3221 from woocommerce/QA-Test-PCP-4318

QA/Add test pcp 4318

mishautkin committed:
Fix typo, file-resolution errors

mishautkin committed:
Add Styling tab functionality and default UI test

- Update Pcp.Admin.Styling types
- Add pcpStyling fixture for admin page
- Add test for default UI
- Update locators in PayPalUI class

Commit:
Add test PCP-4356

mishautkin committed:
Moved methods to snapshot PayPal buttons to PayPalUI class

mishautkin committed:
Merge pull request #3233 from woocommerce/qa-styling

Qa Styling

Commit:
Fix and updates

* subscriptions spec file created
* PCP-4356 moved to subscriptions.spec.ts

Commit:
Fİx lint issues

Commit:
Merge pull request #3234 from woocommerce/QA-add-test-pcp-4356

Qa Add test pcp 4356

mishautkin committed:
Introduce PcpApi + cleanup

- Add class and fixture PcpApi - replaces UI with API methods
- Clean UI methods from utils fixture
- Replace utils UI methods with PcpApi methods
- Fix comments + Lint:js:fix

mishautkin committed:
Remove unused imports from `PcpApi`

mishautkin committed:
Merge pull request #3247 from woocommerce/qa-pcp-api

QA Introduce PcpApi + cleanup of UI methods

Commit:
spec file updated

spec file updated according to latest PR containing pcpApi

Commit:
Spec file renamed

visual-default-ui spec file renamed to visual-onboarding-ui

Commit:
Test keys updated

Test keys for badge values test cases updated

Commit:
Fix badges test

- Germany moved to a separate test block as it has only business account
- Related data removed from ui-test-per-country data file
- countryCode change as country and values updated as countries' name

Commit:
fix country

mishautkin committed:
Add BaseExtend type to playwright.config

mishautkin committed:
Introduce default country/currency based on `.env`

mishautkin committed:
Updated types in Pcp namespace

mishautkin committed:
Fix type in PcpApi updatePaymentMethods

mishautkin committed:
Update -pcp-gateways according to change in types

mishautkin committed:
Merge pull request #3323 from woocommerce/qa-general-updates

Qa general updates

mishautkin committed:
Merge pull request #3324 from woocommerce/qa-pcp-types-update

Qa pcp types update

mishautkin committed:
Split PayPanUi locators and methods

- Separate block locators and methods
- Separate classic locators and methods
- Separate popup locators and methods

mishautkin committed:
Correct fixture naming for PayPalUi, PayPalUiClassic and PayPalApi in test.ts

mishautkin committed:
Update naming of PayPalUi, PayPalUiClassic and PayPalApi in other files.

mishautkin committed:
Fix import in test.ts

mishautkin committed:
Add PcpPaymentMethods class and spec

mishautkin committed:
Update README.md

mishautkin committed:
Merge pull request #3326 from woocommerce/qa-paypal-ui

Qa paypal UI

mishautkin committed:
Update types

- Replaced Pcp.Api with Pcp.Api.PaymentMethods
- Added Pcp.Api.Settings

mishautkin committed:
Add types to makeOrder methods

mishautkin committed:
Add 'Deactivate Disable new UI plugin' to `woocommerce.setup`

mishautkin committed:
Fix paypal popup login with phone

mishautkin committed:
Add transaction tests scenarios for all pages

mishautkin committed:
Add transaction tests initial data for paypal and venmo

mishautkin committed:
Add transaction spect for block and classic pages

Commit:
fixes and updates:
* beforeAll block added
* page titles updated for ES

Commit:
Merge pull request #3333 from woocommerce/sedat-work

fixes and updates

mishautkin committed:
Merge pull request #3329 from woocommerce/qa-type-fix

Qa type fix

mishautkin committed:
Merge pull request #3330 from woocommerce/qa-transaction-tests

Qa transaction tests

Commit:
PCP-4366-4367-4368-4369-4370-4371-4372-4373 added

mishautkin committed:
Migrate tests for legacy UI

- Tests migration
- Tests correction
- Folder structure and file naming fixes

mishautkin committed:
Merge pull request #3353 from woocommerce/qa-legacy-ui

Migrate tests for legacy UI

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa-payment-methods

mishautkin committed:
Update types and pcp-gateways according to requirements

mishautkin committed:
Update test Settings - USA - Payment Methods - Default UI

mishautkin committed:
Lint:js:fix, general fixes (typos, comments, etc.)

mishautkin committed:
Merge pull request #3327 from woocommerce/qa-payment-methods

QA Add PcpPaymentMethods class and spec

mishautkin committed:
Update playwright.config.ts

Reduced test timeout time to 1 min.

mishautkin committed:
Update woocommerce.setup.ts

Uncommented preconditions.

Commit:
Merge branch 'qa' into QA-add-test-pcp-4366-4367-4368-4369-4370-4371-4372-4373

Commit:
fix: PCP-4366,4367,4368,4369,4370,4371,7372,4373

function for screenshot and a locator removed from pcp-onboarding

mishautkin committed:
Update .gitignore

mishautkin committed:
Update .gitignore

mishautkin committed:
Add locators for Onboarding page

mishautkin committed:
Add locators for Onboarding page

mishautkin committed:
Add/update snapshot methods in `pcp-admin-page`

mishautkin committed:
Add/update snapshot methods in `pcp-admin-page`

mishautkin committed:
Remove `percy` from resources

mishautkin committed:
Remove `percy` from resources

mishautkin committed:
Replace `percy` in subscriptions

mishautkin committed:
Replace `percy` in subscriptions

mishautkin committed:
Update `visual-onboarding-ui.spec`

- Replace percy
- Move out badges tests

mishautkin committed:
Update `visual-onboarding-ui.spec`

- Replace percy
- Move out badges tests

mishautkin committed:
Add separate spec with badges tests

mishautkin committed:
Add separate spec with badges tests

mishautkin committed:
General changes

mishautkin committed:
General changes

mishautkin committed:
Merge pull request #3361 from woocommerce/qa-replace-percy

Qa replace percy

mishautkin committed:
Update Playwright Screenshots functionality

- Update options argument and type in pcp-admin-page.snapshotLocator
- Update snapshot methods in PLM, Payment Methods and Styling POMs

mishautkin committed:
Update Playwright Screenshots functionality

- Update options argument and type in pcp-admin-page.snapshotLocator
- Update snapshot methods in PLM, Payment Methods and Styling POMs

Commit:
Merge pull request #3364 from woocommerce/qa-screenshots-correction

QA Update Playwright Screenshots functionality

Commit:
Updates

Extra test block for Germany removed
Test script updated to be dynamic to cover the Germany case
Test data file updated.

Commit:
.only removed

Commit:
Merge branch 'qa' into QA-add-test-pcp-4366-4367-4368-4369-4370-4371-4372-4373

Commit:
Fix and changes

Country and testKey added to test data
Test summary removed
Test script updated accordingly
Fix lint issues

Commit:
Fix and changes

Country and testKey added to test data
Test summary removed
Test script updated accordingly
Fix lint issues

Commit:
Merge pull request #3359 from woocommerce/QA-add-test-pcp-4366-4367-4368-4369-4370-4371-4372-4373

Qa add test pcp 4366 4367 4368 4369 4370 4371 4372 4373

Commit:
fix: Locator name changed and test script updated

Commit:
Tests added:
PCP-4382
PCP-4400

Commit:
fixes and updates

screenshot names updated
page.reload() removed
test titles updated (also on doc)

Commit:
Merge pull request #3375 from woocommerce/QA-Add-test-PCP-4382-and-PCP-4400

QA add test PCP-4382 and PCP-4400

Commit:
Some cosmetic changes

Commit:
QA-add-test-pcp-4403

Commit:
Merge pull request #3381 from woocommerce/QA-add-test-pcp-4403

QA-add-test-pcp-4403

mishautkin committed:
Automation of PCP-4005, PCP-4578, PCP-4596, PCP-3079

- Update type Pcp.Payment
- Add Fastlane payment data
- Add Fastlane guests data
- Add Fastlane test data

mishautkin committed:
Automation of PCP-4005, PCP-4578, PCP-4596, PCP-3079

- Add locators and fastlane methods to PayPalUi classes
- Add locators, fastlane methods and update makeOrder scenario in Checkout classes
- Add fastlane tests to transaction specs

Commit:
QA-add-test-pcp-4357

Test added:
PCP-4357

Commit:
fix: unused fixtures removed

Commit:
fix: remove unused import

mishautkin committed:
Merge pull request #3383 from woocommerce/qa-fastlane

Qa fastlane

Commit:
fix: click action for continueButton moved to test from function

mishautkin committed:
Merge pull request #3384 from woocommerce/QA-add-test-pcp-4357

Qa add test pcp 4357

Commit:
.only removed

mishautkin committed:
Replaced 02-dashboard-ui with 02-onboarding, 03-plugin-settings

mishautkin committed:
Move merchant credentials from `/resources/` to `.env`

mishautkin committed:
Add `.setup` to setup files

mishautkin committed:
Add `.scenario` to files in `_test-scenarios` folders

mishautkin committed:
Add ShopOrder type (has `title` field)

mishautkin committed:
Lint fix

mishautkin committed:
Remove unused methods

mishautkin committed:
Add missing types and descriptions

mishautkin committed:
Merge pull request #3387 from woocommerce/qa-refactoring

Qa refactoring

Commit:
screenshot function changed to snapshotLocator()

mishautkin committed:
Add ACDC tests

- Unify locators and makePayment for block and classic checkout.
- Add ACDC test data.
- Fix assertion in PayPalApi to satisfy both block and classic checkouts.
- Fix cards data.
- Fix ACDC 3DS gateway data.

mishautkin committed:
General fixes, adding missing descriptions, removed @Critical labels

Commit:
Add PCP-4360 and various updates

Commit:
fix: snapshotName var removed

mishautkin committed:
Merge pull request #3393 from woocommerce/qa-acdc

QA Add ACDC tests

mishautkin committed:
Fix `pcpOnboarding.snapshotContent`

Commit:
Merge pull request #3398 from woocommerce/QA-add-test-pcp-4360

QA Add PCP-4360 and various updates

Commit:
Fix 'pcpOnboarding.snapshotLocator' for pcp-4403

Commit:
Fix learn more links and page title texts

mishautkin committed:
QA Add refund tests PCP-1394, PCP-1395, PCP-1405, PCP-1406

- Add ShopRefund type.
- Add refund.paypal.data.
- Add refund.scenario.
- Add refund-usa.spec.
- Update WooCommerceOrderEdit.assertRefundData method.

mishautkin committed:
Merge changes from `qa`

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa-refund

mishautkin committed:
Update playwright-utils version

Commit:
Test added: PCP-4733

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa-refund

mishautkin committed:
Merge pull request #3409 from woocommerce/qa-refund

Qa refund tests PCP-1394, PCP-1395, PCP-1405, PCP-1406

Commit:
Merge pull request #3414 from woocommerce/qa-add-test-pcp-4733

QA Test added: PCP-4733

mishautkin committed:
Update for PayPal user creds

- Move user creds to .env
- Update .env.example
- Update README

mishautkin committed:
Merge pull request #3419 from woocommerce/qa-paypal-creds

QA Update PayPal test user creds

mishautkin committed:
Implement onboarding API options

- Add type Pcp.Api.OnboardingOptions.
- Add onboardingOptions to method PcpApi.connectMerchant.

mishautkin committed:
Merge pull request #3420 from woocommerce/qa-onboarding-api

QA Implement onboarding API options

mishautkin committed:
Move test credentials to `.env` for qa-legacy-ui

mishautkin committed:
Update README.md

mishautkin committed:
Update README.md

mishautkin committed:
Update README.md

mishautkin committed:
Update README.md

mishautkin committed:
Update README.md

mishautkin committed:
Update README.md

mishautkin committed:
Update README.md

mishautkin committed:
Implement vaulting functionality for new UI

- Add second visa
- Fix types
- Split PayPalUi.openPayPalGatewayPupup into separate for PayPal, Pay Later, Venmo.
- Add vaulting functionality to PayPalUi.makePayment.
- Move savepaymentMethod from PayPalUi to CustomerAccount
- Separate completeAcdcPayment method for PayPalUi and PayPalUiClassic.
- Implement PayPalUi.completeAcdcVaultedPayment method.
- Add assertions for not/display of vaulted payment methods to PayPalUi/Classic, CustomerAccount.
- Update locators.

mishautkin committed:
Add vaulting tests for My Account

- 6 test scenarios to save/delete payment methods
- Add README for vaulting tests

mishautkin committed:
Add test scenarios for transactions (see README from previous commit)

mishautkin committed:
Add pay-by-link scenarios

mishautkin committed:
Add tests data

mishautkin committed:
Add spec files

mishautkin committed:
Lint fixes

mishautkin committed:
Merge pull request #3434 from woocommerce/qa-vaulting

QA Implement vaulting functionality for new UI

mishautkin committed:
Cleanup

- Remove unused setup files.
- Move invalid merchant credentials to .env.
- Fix storeConfigGermany in resources/woocommerce-config

mishautkin committed:
Add README for subscription tests.

mishautkin committed:
Update README.md

mishautkin committed:
Add env setup commands

mishautkin committed:
Fix env setup scripts

- Fix command in package.json
- Add reset-sse.sh script
- Fix store.setup amd utils script
- Update README

Commit:
A verificiation has been added for Intent Authorization

Commit:
Comment removed from woocommerce-edit fixture and test commit to see if commit sign works

sedatbsr committed:
Merge pull request #3470 from woocommerce/PCP-4890-transaction-intent-authorize-adding-automated-script-for-capturing-authorized-order

QA - A verification method has been added for Intent Authorization

mishautkin committed:
Update automated env setup script

- Remove reset:sse script
- rename subscription script

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa-subscriptions

mishautkin committed:
Merge pull request #3461 from woocommerce/qa-env-setup

Qa env setup

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa-subscriptions

mishautkin committed:
Update playwright-utils to v2.6.2

mishautkin committed:
Optimize PayPal functionality for subscriptions

- Update locators in paypal-ui.
- Optimize locators and methods in paypal-popup.

mishautkin committed:
Update `/resources/`

- Add data for merchant without reference transactions.
- Update subscription products.
- Add disable webhook verification plugin for PayPal subscriptions.

mishautkin committed:
Update shipping methods selection on checkout pages

mishautkin committed:
Add urls to work with subscriptions

mishautkin committed:
Fix method in customer-payment-methods

mishautkin committed:
Add creation of PayPal subscription products

mishautkin committed:
Add automated env setup script for PayPal subscriptions

mishautkin committed:
Update Readme for automated env setup scripts

mishautkin committed:
Add API methods for subscription renewals

mishautkin committed:
Add subscription transaction tests for checkout pages

- Add tests data
- Add test scenarios
- Add test specs
- Update README

mishautkin committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce-paypal-payments into qa

mishautkin committed:
Merge pull request #3450 from woocommerce/qa-subscriptions

QA. Subscription tests.

mishautkin committed:
Fix PayPal popup

- Add try methods
- Update locators

mishautkin committed:
Merge pull request #3492 from woocommerce/qa-fix-paypal-popup

QA. Fix PayPal popup

mishautkin committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce-paypal-payments into qa

mishautkin committed:
Update playwright-utils version

mishautkin committed:
Add functionality for subscription renewal

- Add API methods to work with subscriptions.
- Fix wooCommerceOrderEdit method.
- Remove assertSubscriptionDetails from wooCommerceSunscriptionEdit since it's present in parent class.
- Fix method name in utils.

mishautkin committed:
Add subscription renewal data, scenario, spec, README

mishautkin committed:
General fixes in package.json and playwright.config

mishautkin committed:
Merge pull request #3507 from woocommerce/qa-subscriptions

Qa subscriptions

mishautkin committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce-paypal-payments into qa

Commit:
couponMessage text updated

mishautkin committed:
General updates

- update playwright-utils version.
- update timeout in playwright config

mishautkin committed:
Fix paypal popup behavior

mishautkin committed:
Update default country to USA

mishautkin committed:
Move specs from Germany to USA as default country

mishautkin committed:
Fix assertions according to new changes

mishautkin committed:
Update test title

mishautkin committed:
Lint fix

mishautkin committed:
Remove unused methos (PayPalUi -> savePaymentMethod)

mishautkin committed:
Fix for DDEV

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa

mishautkin committed:
General updates

- Update playwright-utils version
- Update test timeout
- Update gitignored playwright-report dir
- Rename this.popup to this.page in PayPalPopup.

mishautkin committed:
Update resources

- Update prices for products
- Update guest for fastlane
- Add/remove missing/extra test data
- Add installation of webhook verification plugin

mishautkin committed:
Fix transaction pre-conditions/post-conditions

- Add afterAll to remove the subscriptions and orders
- Add preconditions to fix the restoration of the customer - dependencies need to be removed - customer's orders, subscriptions, user.

mishautkin committed:
Add test keys, links and bug to test data

mishautkin committed:
Fix PcpApi.getSubscriptionRenewalOrderIds

mishautkin committed:
Fix utils.restoreCustomer

Add removal of related user. It prevented from creating customer with the same email.

mishautkin committed:
Fix CustomerPaymentMethods.savePaymentMethod

mishautkin committed:
Fix locators and methods in PayPalUi

mishautkin committed:
Fix assertion in PayPalUiClassic

mishautkin committed:
Update subscription-renewal.spec

- Reuse disableWebhookVerifivationPlugin.slug from resources

mishautkin committed:
Fix extra export

mishautkin committed:
Fixes for PCP v3.0.8

stracker-phil commented on tests/qa-legacy-ui/utils/admin/connection.ts:
```diff
@@ -0,0 +1,206 @@
+		this.page.getByRole( 'link', { name: 'Get Help', exact: true } );
+
+	enablePayPalPaymentsCheckbox = () =>
+		this.page.getByLabel(
```
We should implement `data-testid` attributes for all UI elements that are used for tests

A first step could be searching test cases for `getByLabel()` and update the relevant plugin code elements to provide a unique `data-testid` attribute, so we can gradually move away from using labels in test cases

mishautkin replied:
Thanks, @stracker-phil, tha would be a great leap ))

puntope commented on tests/qa/README.md:
```diff
@@ -0,0 +1,348 @@
+
+## Installation of `node_modules`
+
+1. Remove `"workspaces": [ "playwright-utils" ]` from `package.json`.
```
I don't see this in package.json

mishautkin replied:
@puntope, for usage of `playwright-utils` as a package this could be rephrased as follows:

> 1. Make sure `"workspaces": [ "playwright-utils" ]` aren't present in package.json.

It shouldn't be present unless one's going to develop code in `playwright-utils` package. This solution was inplemented by Moritz (@Biont) and unfortunately I haven't been able to find a successful workaround yet (like `npm link` or similar).

mishautkin replied:
@puntope updated in readme, please check.

puntope commented on tests/qa/README.md:
```diff
@@ -0,0 +1,348 @@
+
+## Project configuration
+
+1. [SSE setup](https://inpsyde.atlassian.net/wiki/spaces/AT/pages/3175907370/Self+Service+WordPress+Environment) - will be deprecated in Q1 of 2025.
```
This seems archived already. It's still valid?

mishautkin replied:
Was still in use before my vacation 3 weeks ago. For that time Kinsta was protected by 2FA and I couldn't run tests there..

puntope commented on tests/qa/README.md:
````diff
@@ -0,0 +1,348 @@
+2. In the test project (`./tests/qa/`) run following command:
+
+```bash
+npm run setup:tests
````
This script failed for me:

```
$ npm run setup:tests

> woocommerce-paypal-payments-tests@1.0.0 setup:tests
> npm install && npx playwright install

npm error code E401
npm error 401 Unauthorized - GET https://npm.pkg.github.com/@inpsyde%2fplaywright-utils - authentication token not provided
npm error A complete log of this run can be found in: 
```

AlexP11223 replied:
You can add the auth token somewhere, but yeah, I think all packages here should be public.

puntope replied:
In my case fixed by running `npm login --registry=https://npm.pkg.github.com --scope=@inpsyde` 

Maybe we can document this as well in the README setup section

mishautkin replied:
@puntope It is a general approach for Syde packages. It is actually documented [in playwright-utils Readme](https://github.com/inpsyde/playwright-utils?tab=readme-ov-file#package-requirements) but I can also include it here.

mishautkin replied:
@puntope updated in readme, please check.

puntope commented on tests/qa/README.md:
```diff
@@ -0,0 +1,348 @@
+
+	2.1 Set general variables following [these steps](https://github.com/inpsyde/playwright-utils?tab=readme-ov-file#env-variables).
+	
+	2.2 Set PayPal API keys and test credentials. See `.env.example`. The `.env` content with actual test users' credentials is [stored in 1Password](https://start.1password.com/open/i?a=UL7QZZ6P6JDVBI422AOVJXMEGU&v=qrw2rcy27xpwiedd5gifyga2pa&i=iddsl6wdm62lwwqbzz474jodme&h=inpsyde.1password.eu).
```
I cannot find this

mishautkin replied:
I have to check. I think that @Dinamiko was able to access that vault, so I thought it is visible within the company. We can check it.

mishautkin replied:
@puntope updated in readme, please check if you can access the vault now.

puntope commented on tests/qa-legacy-ui/tests/01-plugin-foundation/plugin-foundation.spec.ts:
```diff
@@ -0,0 +1,65 @@
+/**
+ * Internal dependencies
+ */
+import { test, expect } from '../../utils';
```
Imports for test, expect and  customers are not used.

puntope commented on tests/qa-legacy-ui/tests/01-plugin-foundation/plugin-foundation.spec.ts:
```diff
@@ -0,0 +1,65 @@
+import {
+	testPluginInstallationFromFile,
+	testPluginReinstallationFromFile,
+	testPluginInstallationFromMarketplace,
```
Note that  there is a commented critical test using this one

puntope commented on tests/qa-legacy-ui/tests/01-plugin-foundation/plugin-foundation.spec.ts:
```diff
@@ -0,0 +1,65 @@
+
+testPluginReinstallationFromFile( 'PCP-1007', pcpPlugin, '@Critical' );
+
+// testPluginInstallationFromMarketplace( 'PCP-1004', pcpPlugin, '@Critical' );
```
Why is this commented?

puntope commented on tests/qa/tests/01-plugin-foundation/plugin-foundation.spec.ts:
```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
```
Some leftover commented code here.

mishautkin replied:
Removed.

puntope commented on tests/qa/tests/03-plugin-settings/pay-later-messaging-configurator.spec.ts:
```diff
@@ -0,0 +1,351 @@
+/**
```
Some leftover commented code and ToDos in this file.

mishautkin replied:
Will be removed if/when the UI is fixed, as stated in comments

puntope commented on tests/qa/resources/disable-nonce-plugin.json:
```diff
@@ -0,0 +1,7 @@
+  "name": "Disable nonce check",
+  "slug": "disable-nonce-check",
+  "zipFile": "disable-nonce.zip",
+  "zipFilePath": "./resources/files/disable-nonce.zip",
```
Most of these resource plugins are giving me errors. Not sure if I need to install them on my site or they should be committed in the files folder as `"zipFilePath": "./resources/files/disable-nonce.zip", `  indicates.

mishautkin replied:
@puntope I raised the problem several times but we haven't fount the solution yet. The .zip files are currently gitignored in the repo so they can't be commited. On the other hand we need some of simple one-line plugins to apply some filters during autotesting as well as bigger ones like WooCommerce Subscriptions (can't be installed from the market because it isn't free).

Any suggestions?
CC @Dinamiko @stracker-phil @AlexP11223

AlexP11223 replied:
`git add -f my.zip`?

Or unignoring, something like `!dir/with/zips/*.zip` in `.gitignore`

mishautkin replied:
@AlexP11223, Thanks. Technically that's clear. I mean, since this isn't our internal repo, I thouhgt that I should first agree with someone that I can unignore .zip files in `/tests/qa`. If it's that simple - I will solve the problem easily ))

AlexP11223 replied:
Well, the `.gitignore` was created by us. It ignores zips to ignore the plugin builds, when it was built locally.

mishautkin replied:
Ok, @AlexP11223, in this case I'll include the files. Thank you

mishautkin replied:
Un-gitignored `/tests/qa/resources/files/*.zip` (also in `/tests/qa-legacy-ui/`)

puntope commented:
⚠️ Some failed tests mostly in store.setup.ts and woocommerce.setup.ts I assume because of [these problem](https://github.com/woocommerce/woocommerce-paypal-payments/pull/3499#discussion_r2298565344)

```
 13 failed
    [setup-woocommerce] › tests/_setup/woocommerce.setup.ts:29:6 › Setup Disable Nonce plugin (inactive)
    [setup-woocommerce] › tests/_setup/woocommerce.setup.ts:61:6 › Setup Disable Webhook Verification plugin (inactive)
    [setup-woocommerce] › tests/_setup/woocommerce.setup.ts:71:6 › Setup Disable WooCommerce Setup Wizard Plugin (active)
    [setup-woocommerce] › tests/_setup/woocommerce.setup.ts:94:6 › Setup WC Subscriptions plugin (inactive)
    [setup-store] › tests/_setup/store.setup.ts:47:8 › setup:pcp:usa; Setup store settings ─────────
    [setup-store] › tests/_setup/store.setup.ts:47:8 › setup:pcp:germany; Setup store settings ─────
    [setup-store] › tests/_setup/store.setup.ts:64:9 › setup:pcp:germany; Onboarding ───────────────
    [setup-store] › tests/_setup/store.setup.ts:47:8 › setup:pcp:mexico; Setup store settings ──────
    [setup-store] › tests/_setup/store.setup.ts:64:9 › setup:pcp:mexico; Onboarding ────────────────
    [setup-store] › tests/_setup/store.setup.ts:47:8 › setup:pcp:usa:vaulting; Setup store settings
    [setup-store] › tests/_setup/store.setup.ts:47:8 › setup:pcp:usa:vaulting:classic; Setup store settings
    [setup-store] › tests/_setup/store.setup.ts:47:8 › setup:pcp:usa:vaulting:subscription; Setup store settings
    [setup-store] › tests/_setup/store.setup.ts:211:6 › setup:pcp:usa:paypal:subscription; ────────
```

AlexP11223 commented on tests/qa-legacy-ui/package.json:
```diff
@@ -0,0 +1,58 @@
+    "lint:md": "wp-scripts lint-md-docs ./**/*.md README.md",
+    "lint:md:fix": "wp-scripts lint-md-docs --fix ./**/*.md README.md",
+    "lint:js": "wp-scripts lint-js  --resolve-plugins-relative-to ./ **/*.{ts,tsx,mjs}",
+    "lint:js:fix": "wp-scripts lint-js --resolve-plugins-relative-to ./ --fix **/*.{ts,tsx,mjs}",
```
nitpick: there is probably a simpler way without the code duplication, removing `lint:js:fix` and running `npm run lint:js --fix`

mishautkin replied:
@AlexP11223 Thanks for the hint.

mishautkin replied:
@AlexP11223, did you mean this?
```
"lint:js:fix": "npm run lint:js --fix",
```
It didn't work for me.

mishautkin replied:
Leaving for now as it is.

Dinamiko commented:
Just leaving here some notes regarding running the tests locally:
- Regarding Unauthorized error when running `npm run setup:tests`, a personal access token needs to be created in GH and then configured in your machine, ping me for more info how to do it.
- Configuring `.env`
    - Use `http` in `WP_BASE_URL` otherwise it throws error “unable to verify the first certificate” in ddev
    - For `WC_API_KEY` and `WC_API_SECRET` create WooCommerce REST API credentials
- Set WordPress permalinks to “Post name”

Then you can run for example: `npm run test:critical`(`"test:critical": "npx playwright test --grep \"@Critical\""` needs to be added to package.json scripts)
As @puntope commented there are missing .zip files in `resources` so those tests are failing, the rest of critical tests are passing correctly:
```
1) [setup-woocommerce] › tests/_setup/woocommerce.setup.ts:29:6 › Setup Disable Nonce plugin (inactive) 
    Error: ENOENT: no such file or directory, stat './resources/files/disable-nonce.zip'
    
2) [setup-woocommerce] › tests/_setup/woocommerce.setup.ts:61:6 › Setup Disable Webhook Verification plugin (inactive) 
    Error: ENOENT: no such file or directory, stat './resources/files/disable-webhook-verification.zip' 
    
3) [setup-woocommerce] › tests/_setup/woocommerce.setup.ts:71:6 › Setup Disable WooCommerce Setup Wizard Plugin (active) 
    Error: ENOENT: no such file or directory, stat './resources/files/disable-wc-setup-wizard.zip'   
```

mishautkin committed:
Update README.md

mishautkin committed:
Un-gitignore auxiliary .zip files in tests/qa and tests/qa-legacy-ui

mishautkin committed:
Updated README.md

mishautkin committed:
Lint js and md fixes

mishautkin committed:
General updates

- Update playwright-utils version in package.json
- Uncomment woocommerce.setup

mishautkin committed:
Un-gitignore .zip files in `resources/files`

mishautkin committed:
Merge branch 'qa' of https://github.com/woocommerce/woocommerce-paypal-payments into qa-fix-legacy-ui

mishautkin committed:
Merge pull request #3632 from woocommerce/qa-fix-legacy-ui

Qa fix legacy UI

mishautkin committed:
Update `playwright-utils` to v2.7.2

mishautkin commented:
@Dinamiko @puntope @AlexP11223 I'm done with the changes. Readme are updated, `.zip` files are there now. The scripts work on my side. Could you please review and try out the scripts?

puntope commented:
If the site is MultiSite (the default config for project DDEV env) tests will fail.
Also if I deactivate the multisite in my environment and perform the test against the environment created by the repo the foundation tests fails.  

You can check this error by:

- cloning the repo (using this branch)
- `npm run ddev:setup`
- Remove lines in `.ddev/wordpress/wp-config.php` regarding multisite
- `cd tests/qa`
- `npm run setup:tests`
- `npm run all`

Maybe we can add in the README that the environment set in the `.env` should not be a fresh WordPress Single site and not the one created using `ddev:setup`

puntope commented:
> These 2 criticals dont work for me anymore

```
testPluginInstallationFromFile( 'PCP-1000', pcpPlugin, '@Critical' );
testPluginReinstallationFromFile( 'PCP-1007', pcpPlugin, '@Critical' );
```

puntope commented:
Lots of tests are based on snapshots.  But we didn't commit the snapshots. We should commit these snapshots as well.

puntope commented:
Talking about snapshots again, I'm a bit curious about Snapshot-based E2E testing. What is the rationale behind using them?

puntope commented:
`03-Plugin Settings` Tests are timing out for me.  If they work for you, maybe is something not correctly setup in my env?

puntope commented:
Have we considered using [wp-env](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/) for running the E2E tests?

Here is an [example](https://github.com/woocommerce/google-listings-and-ads/blob/develop/.wp-env.json) how to use wp-env to run the E2E. The good thing about it is that it's ephemeral, and all the devs and also GitHub bots can even use the same standard environment for E2E tests. It's very easy to install (just a dependency) 

You can also load your own plugins or logic easily to manipulate the environment and connect it to WPCLI etc It's widely used in Woo ecosystem 

The same goes for Integration testing 

cc: @Dinamiko

mishautkin committed:
Remove Percy vars from .env.example

mishautkin committed:
Add `.fixme` to 03-plugin-settings tests

mishautkin committed:
Add docs/, refactor main README

mishautkin committed:
Add .fixme to onboarding-badges visual tests

mishautkin committed:
Administrative updates

mishautkin committed:
Un-gitignore screenshots for onboarding tests

mishautkin commented:
Hi, @puntope, I've disabled many visual tests and commited the screenshots for some which are more stable. There are still some failing tests but it would be nice to try running against the `wp-env` or Kinsta, since our current SSEs are a bit slow and may cause problems by themselves.

mishautkin committed:
Add new versions of PCP and WC Subscriptions plugins

puntope reviewed (approved):
Let merge as it's. I can use this as fundational and create the wp-env version from there.

mishautkin committed:
Update playwright-utils to v2.7.3

mishautkin committed:
Fix onboarding-ui.spec

mishautkin committed:
Add comments
